modules/objdetect/misc/python/test/test_charuco_board.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import print_function
+
+import os, tempfile, numpy as np
+
+import sys
+sys.path.append("../../../../../doc/pattern_tools")
+import gen_pattern
+
+import cv2 as cv
+
+from tests_common import NewOpenCVTests
+
+class aruco_objdetect_test(NewOpenCVTests):
+
+    def test_aruco_dicts(self):
+        cols = 3
+        rows = 5
+        square_size = 100
+        aruco_type_ = [cv.aruco.DICT_4X4_1000, cv.aruco.DICT_5X5_1000, cv.aruco.DICT_6X6_1000,
+                       cv.aruco.DICT_7X7_1000, cv.aruco.DICT_ARUCO_ORIGINAL, cv.aruco.DICT_APRILTAG_16h5,
+                       cv.aruco.DICT_APRILTAG_25h9, cv.aruco.DICT_APRILTAG_36h10, cv.aruco.DICT_APRILTAG_36h11]
+        aruco_type_str_ = ['DICT_4X4_1000','DICT_5X5_1000', 'DICT_6X6_1000',
+                       'DICT_7X7_1000', 'DICT_ARUCO_ORIGINAL', 'DICT_APRILTAG_16h5',
+                       'DICT_APRILTAG_25h9', 'DICT_APRILTAG_36h10', 'DICT_APRILTAG_36h11']
+        marker_size = 0.8*square_size
+        board_width = cols*square_size
+        board_height = rows*square_size
+
+        for aruco_type_i in range(len(aruco_type_)):
+            #draw desk using opencv
+            aruco_dict = cv.aruco.getPredefinedDictionary(aruco_type_[aruco_type_i])
+            board = cv.aruco.CharucoBoard((cols, rows), square_size, marker_size, aruco_dict)
+            charuco_detector = cv.aruco.CharucoDetector(board)
+            markerImage = board.generateImage((cols*square_size*10, rows*square_size*10))
+            fd, filepng = tempfile.mkstemp(prefix="cv_marker", suffix=".png")
+            os.close(fd)
+            cv.imwrite("cv_marker.png", markerImage)
+            from_cv_img = cv.imread('cv_marker.png')
```

Comment:
It looks suspicious. Why do you need to save OpenCV rendered board to file and read it back again?